pyairnow/historical.py

```diff
@@ -11,23 +12,32 @@ class Historical:
     def __init__(self, request: Callable[..., Coroutine]) -> None:
         self._request = request
 
+    '''If date is given as string, must be in yyyy-mm-d format. '''
     async def zipCode(
         self,
         zipCode: str,
         *,
-        date: Optional[Union[date_, datetime, str]] = None,
+        date: Union[date_, datetime, str],
         distance: Optional[int] = None
     ) -> list:
         '''Request current observation for zip code'''
         params: dict = dict(zipCode=zipCode)
         '''Airnow  needs T00-0000 appended to the date parameter for historical calls'''
-        if date and isinstance(date, str):
+        if isinstance(date, str):
             y, m, d = date.split('-')
-            params['date'] = date_(int(y), int(m), int(d)).isoformat() + "T00-0000"
-        elif date and isinstance(date, datetime):
-            params['date'] = date.date().isoformat() + "T00-0000"
-        elif date and isinstance(date, date_):
-            params['date'] = date.isoformat() + "T00-0000"
+            '''create a timezone object with no utc offset'''
+            tz = timezone(timedelta())
+            params['date'] = datetime(int(y), int(m), int(d), tzinfo=tz).strftime("%Y-%m-%dT%H%z")
+        elif isinstance(date, datetime):
+            tz = timezone(timedelta())
+            date = date.replace(tzinfo=tz)
+            params['date'] = date.strftime("%Y-%m-%dT%H%z")
+        elif isinstance(date, date_):
+            tz=timezone(timedelta())
+            params['date'] = datetime(date.year, date.month, date.day, tzinfo=tz).strftime("%Y-%m-%dT%H%z")
+        else:
+            raise AirNowError("Bad date type. Date parameter is either date, datetime, or a string.")
```

Comment:
@asymworks I switched to using strftime to get the right date formatting as you suggested. Interestingly, when a datetime object is set to utc timezone, the offset becomes +0000 and not -0000. I tested this with some simple calls to the api and it works just fine either way.